_api-reference/ingest-apis/ingest-pipelines.md

```diff
@@ -0,0 +1,382 @@
+            }
+          }
+        },
+        {
+          "processor_type": "uppercase",
+          "status": "success",
+          "doc": {
+            "_index": "my-index",
+            "_id": "1",
+            "_source": {
+              "name": "JOHN DOE",
+              "grad_year": 2023,
+              "graduated": true
+            },
+            "_ingest": {
+              "pipeline": "my-pipeline",
+              "timestamp": "2023-06-20T23:23:26.656564631Z"
+            }
+          }
+        }
+      ]
+    }
+  ]
+}
+```
+
+### Example: Specify a pipeline in the request body
+
+Alternatively, you can specify a pipeline directly in the request body without creating a pipeline first:
+
+```json
+POST /_ingest/pipeline/_simulate
+{
+  "pipeline" :
+  {
+    "description": "Splits text on whitespace characters",
+    "processors": [
+      {
+        "csv" : {
+          "field" : "name",
+          "separator": ",",
+          "target_fields": ["last_name", "first_name"],
+          "trim": true
+        }
+      },
+      {
+      "uppercase": {
+        "field": "last_name"
+      }
+    }
+    ]
+  },
+  "docs": [
+    {
+      "_index": "second-index",
+      "_id": "1",
+      "_source": {
+        "name": "Doe,John"
+      }
+    },
+    {
+      "_index": "second-index",
+      "_id": "2",
+      "_source": {
+        "name": "Doe, Jane"
+      }
+    }
+  ]
+}
+```
+
+## Get information about a pipeline
+
+Use the get ingest pipeline API operation to return information about a pipeline.
+
+### Retrieving information about all pipelines
+
+The following example request returns information about all ingest pipelines:
+
+```json
+GET _ingest/pipeline/
+```
+{% include copy-curl.html %}
+
+### Retrieving information about a specific pipeline
+
+The following example request returns information about a specific pipeline: 
+
+```json
+GET _ingest/pipeline/my-pipeline
+```
+{% include copy-curl.html %}
+
+The response contains the pipeline information:
+
+```json
+{
+  "my-pipeline": {
+    "description": "This pipeline processes student data",
+    "processors": [
+      {
+        "set": {
+          "description": "Sets the graduation year to 2023",
+          "field": "grad_year",
+          "value": 2023
+        }
+      },
+      {
+        "set": {
+          "description": "Sets graduated to true",
+          "field": "graduated",
+          "value": true
+        }
+      },
+      {
+        "uppercase": {
+          "field": "name"
+        }
+      }
+    ]
+  }
+}
+```
+
+## Delete a pipeline
+
+Use the following requests to delete pipelines. 
+
+### Delete a specific pipeline
+
+```json
+DELETE /_ingest/pipeline/<pipeline-id>
+```
+{% include copy-curl.html %}
+
+### Delete all pipelines
+
+```json
+DELETE /_ingest/pipeline/*
+```
+{% include copy-curl.html %}
+
+
+## Create a pipeline on an existing index
+
+_<Do we want to include a section on creating a pipeline on an existing index?>
```

Comment:
@heemin32 Do we want to include a section about creating a pipeline on an existing index?

Reply:
As long as I know, there is no such thing like creating a pipeline on an existing index.
Once we create a ingest pipeline, we can use the pipeline to add a document to an existing index.